torchvision/prototype/transforms/_transform.py

```diff
@@ -12,6 +12,10 @@ def __init__(self) -> None:
         super().__init__()
         _log_api_usage_once(self)
 
+    # FYI: I spoke with @NicolasHug offline and provided a valid use-case where it would be useful to make this public,
+    # as you originally intended. We don't have to do this now but if we do end up exposing this publicly, we should
+    # rename it to avoid conflicts with the previous static get_params() which worked completely differently.
+    # We dn't have to do anything now but we should discuss this again after the feedback.
```

Comment:
Could you share this use case with us? cc @NicolasHug

Reply:
I think the use-case is this: https://github.com/pytorch/vision/pull/3001#issuecomment-814849595

I.e. the ability to "freeze" a transform that has randomly-sampled parameters. This isn't too exotic of a need, I has the same use-case in the RAFT training reference, and ended having to workaround it by batching images (but in some scenarios batching isn't possible if all images aren't available at the same time)

https://github.com/pytorch/vision/blob/95d418970e6dbf2e4d928a204c4e620da7bccdc0/references/optical_flow/transforms.py#L94-L97

Reply:
IIUC, the use case is to transform multiple images with the same randomly sampled parameters, correct? If yes, this already works:

https://github.com/pytorch/vision/blob/95d418970e6dbf2e4d928a204c4e620da7bccdc0/torchvision/prototype/transforms/_transform.py#L22-L23

We sample the parameters once for the complete input and afterwards apply the transformation with these sampled parameters to all inputs individually. For example, let's say you want to randomly rotate two images the same way:

```py
transforms = RandomRotate(...)

image1, image2 = transform(image1, image2)
```

This of course also means we need no extra handling for image pair datasets:

```py
transform = RandomRotate(...)
dataset = datasets.load("sintel").map(transform)

for sample in dataset:
    ...
```

Reply:
Is it still supported when images are not available at the same time? i.e.

```
t = SomeRandomTransform(...)
t(img1)
t(img2)
...
t(imgn)
```

?

Reply:
@NicolasHug Hmmm... No it's not. Can you describe more of why this is the case?

Reply:
In general, *some* users may need to call the exact same transform (with the same randomly-sampled parameters) on all the batches during a single iteration. And all the batches aren't available at the same time.

However as we just discussed offline, I don't have a clear, immediate use-case for these, apart from the link to the PR above. In practice the one use-case that I encountered (the RAFT thing above) is properly covered by the design as @pmeier indicated https://github.com/pytorch/vision/pull/5500#discussion_r816615926 .

Since our current transforms don't support that anyway, we're not introducing any regression. And since most of the details are kept private, we should be able to support it in some way in the future, if we ever find a stronger need for it.

But for now I'm happy to keep things as they are.